src/AsmResolver.DotNet/Signatures/GenericContext.cs

```diff
@@ -18,39 +19,45 @@ public GenericContext(IGenericArgumentsProvider? type, IGenericArgumentsProvider
             Type = type;
             Method = method;
         }
-        
+
         /// <summary>
-        /// Gets the object responsible for providing type arguments defined by the current generic type instantiation. 
+        /// Gets the object responsible for providing type arguments defined by the current generic type instantiation.
         /// </summary>
         public IGenericArgumentsProvider? Type
         {
             get;
         }
 
         /// <summary>
-        /// Gets the object responsible for providing type arguments defined by the current generic method instantiation. 
+        /// Gets the object responsible for providing type arguments defined by the current generic method instantiation.
         /// </summary>
         public IGenericArgumentsProvider? Method
         {
             get;
         }
 
         /// <summary>
-        /// Enters a new generic context with a new type providing type arguments. 
+        /// Returns true if both Type and Method providers are null
+        /// </summary>
+        public bool IsEmpty => Type is null && Method is null;
```

Comment:
Add `[MemberNotNullWhen]` attributes referencing both the `Type` and `Method` properties.

Reply:
Not relevant since we don't know which member is not null